wallet/src/main/scala/org/bitcoins/wallet/api/SerializedTransaction.scala

```diff
@@ -0,0 +1,177 @@
+}
+import org.bitcoins.core.script.constant.{
+  ScriptConstant,
+  ScriptNumberOperation,
+  ScriptToken
+}
+import play.api.libs.json.{JsNumber, JsString, Json, Writes}
+import scodec.bits.ByteVector
+
+case class SerializedTransaction(
+    txid: DoubleSha256DigestBE,
+    wtxid: Option[DoubleSha256DigestBE],
+    version: Int32,
+    size: Long,
+    vsize: Long,
+    weight: Long,
+    locktime: UInt32,
+    vin: Vector[SerializedTransactionInput],
+    vout: Vector[SerializedTransactionOutput])
+
+case class SerializedTransactionInput(
+    txid: DoubleSha256DigestBE,
+    hex: String,
+    vout: UInt32,
+    scriptSig: Vector[ScriptToken],
+    txinwitness: Option[SerializedTransactionWitness],
+    sequence: UInt32
+)
+
+case class SerializedTransactionWitness(
+    hex: String,
+    scriptType: Option[String],
+    script: Option[Vector[ScriptToken]],
+    pubKey: Option[ECPublicKey],
+    signature: Option[ECDigitalSignature],
+    stack: Option[Vector[ByteVector]])
+
+case class SerializedTransactionOutput(
+    value: BigDecimal,
+    n: UInt32,
+    scriptPubKey: ScriptPubKey,
+    hex: String
+)
+
+object SerializedTransaction {
+
+  def decodeRawTransaction(tx: Transaction): String = {
+    def tokenToString(token: ScriptToken): String = {
+      token match {
+        case numOp: ScriptNumberOperation => numOp.toString
+        case constOp: ScriptConstant      => constOp.bytes.toString
+        case otherOp                      => otherOp.toString
+      }
+    }
+
+    implicit val byteVectorWrites: Writes[ByteVector] =
+      Writes[ByteVector](bytes => JsString(bytes.toHex))
+    implicit val ecDigitalSignatureWrites: Writes[ECDigitalSignature] =
+      Writes[ECDigitalSignature](sig => JsString(sig.hex))
+    implicit val ecPublicKeyWrites: Writes[ECPublicKey] =
+      Writes[ECPublicKey](pubKey => JsString(pubKey.hex))
+    implicit val scriptTokenWrites: Writes[ScriptToken] =
+      Writes[ScriptToken](token => JsString(tokenToString(token)))
+    implicit val doubleSha256DigestBEWrites: Writes[DoubleSha256DigestBE] =
+      Writes[DoubleSha256DigestBE](hash => JsString(hash.hex))
+    implicit val uInt32Writes: Writes[UInt32] =
+      Writes[UInt32](num => JsNumber(num.toLong))
+    implicit val int32Writes: Writes[Int32] =
+      Writes[Int32](num => JsNumber(num.toLong))
+    implicit val scriptPubKeyWrites: Writes[ScriptPubKey] =
+      Writes[ScriptPubKey](spk => Json.toJson(spk.asm))
+
+    implicit val serializedTransactionWitnessWrites: Writes[
+      SerializedTransactionWitness] = Json.writes[SerializedTransactionWitness]
+    implicit val serializedTransactionInputWrites: Writes[
+      SerializedTransactionInput] = Json.writes[SerializedTransactionInput]
+    implicit val serializedTransactionOutputWrites: Writes[
+      SerializedTransactionOutput] = Json.writes[SerializedTransactionOutput]
+    implicit val serializedTransactionWrites: Writes[SerializedTransaction] =
+      Json.writes[SerializedTransaction]
```

Comment:
Maybe move these to the bottom of the page, so you don't need to scroll down as much to get to the actual code